### ISSUE TITLE
TypeScript build failing in Node.js-only environment

### ISSUE THREAD

**Issue by Jomshir98** [socketio/socket.io-client#1561]
**Describe the bug**
When building project using this as dependency using typescript with following option (no DOM library):
```
"compilerOptions": {
    "lib": ["ES2020"], // Or anything really that doesn't include "dom" as one of options
}
```
The project fails to build with following errors:
```
node_modules/engine.io-client/build/esm/transport.d.ts:12:15 - error TS2304: Cannot find name 'CloseEvent'.

12     context?: CloseEvent | XMLHttpRequest;
                 ~~~~~~~~~~

node_modules/engine.io-client/build/esm/transport.d.ts:12:28 - error TS2304: Cannot find name 'XMLHttpRequest'.

12     context?: CloseEvent | XMLHttpRequest;
                              ~~~~~~~~~~~~~~

node_modules/engine.io-client/build/esm/transports/polling.d.ts:91:43 - error TS2304: Cannot find name 'XMLHttpRequest'.

91     error: (err: number | Error, context: XMLHttpRequest) => void;
                                             ~~~~~~~~~~~~~~

node_modules/socket.io-client/build/esm/socket.d.ts:14:15 - error TS2304: Cannot find name 'CloseEvent'.

14     context?: CloseEvent | XMLHttpRequest;
                 ~~~~~~~~~~

node_modules/socket.io-client/build/esm/socket.d.ts:14:28 - error TS2304: Cannot find name 'XMLHttpRequest'.

14     context?: CloseEvent | XMLHttpRequest;
                              ~~~~~~~~~~~~~~


Found 5 errors in 3 files.
```

The same problem also affects engine.io, as seen.

**Context**
Triaged this issue to version 4.5.0, likely following commit: b862924b7f1720979e5db2f0154906b305d420e3

The error is because in node-only environment no `CloseEvent` nor `XMLHttpRequest` is defined.

**Fix suggestion**
Similar issues in the past in typings have been fixed by adding following comment to the typings file:
```
/// <reference lib="dom" />
```

Temporary workaround for anyone else facing this issue is adding following to TS config:
```
"compilerOptions": {
    "skipLibCheck": true
}
```

**Comment by callum-bennett** [comment]
Adding "dom" to the compilerOptions lib array also fixes this issue.

```
  "compilerOptions": {
    "lib": ["es2020", "dom"],
  },
```

**Comment by Jomshir98** [comment]
> Adding "dom" to the compilerOptions lib array also fixes this issue.
> 
> ```
>   "compilerOptions": {
>     "lib": ["es2020", "dom"],
>   },
> ```

Yes, I literally said in the issue that making Node.js only environment (which means not having DOM) is what causes this. So adding "dom" is not a solution, as it makes TS think the code is to run in a browser, not in Node.js.
This issue is about Node.js-only applications where we want TS to show error whenever someone tries to use something Node doesn't have.

**Comment by darrachequesne** [comment]
I could indeed reproduce the issue, let me check.

**Comment by darrachequesne** [comment]
This should be fixed in version [4.6.0](https://github.com/socketio/socket.io-client/releases/tag/4.6.0) (see https://github.com/socketio/socket.io-client/commit/4d6d95e0792efd43b78c760b055764fef02ebc9e).

Have a great day!
